porchlight/neighborhood.py

```diff
@@ -426,6 +457,213 @@ def call_all_doors(self):
 
                 self._params[pname].value = new_value
 
+    def call(self, door_name: Union[str, door.Door]) -> Any:
+        """Executes a call for a single door."""
+        # Check if ids need to be checked.
+        if not isinstance(door_name, str) and isinstance(door_name, door.Door):
+            if not any(door_name is d for d in self._doors.values()) or not any(
+                door_name.name == d for d in self._doors
+            ):
+                msg = (
+                    f"Could not find door {door_name.name} in Neighborhood."
+                    f" The following doors are present:\n"
+                )
+
+                msg += "; ".join(d for d in self.doors)
+
+                logging.error(msg)
+                raise ValueError(msg)
+
+            door_name = door_name.name
+
+        elif not isinstance(door_name, str):
+            msg = (
+                f"Expected a string or door as input, got {door_name} "
+                f"({type(door_name)})"
+            )
+
+            logging.error(msg)
+            raise TypeError(msg)
+
+        # TODO: I think this needs to have a way to not add any new paramaters
```

Comment:
Can this be done before this pull?

Reply:
After playing around with this and implementing it I think this is better suited for 2.0, when some other feature concepts have fully developed.